### ISSUE TITLE
Could not get it to work on Fedora 34 (Gnome 40)

### ISSUE THREAD

**Issue by taratatach** [fabi1cazenave/qwerty-lafayette#90]
Since I migrated to fedora 34, which includes Gnome 40, I can't find Lafayette in the Gnome keyboard selector and `setxkbmap` seems to work partially in Alacritty (no altgr layer).

**Comment by JulioJu** [comment]
Why do you use Gnome ? 😄 :trollface:

**Comment by taratatach** [comment]
So I got it to mostly work on some applications with the following command:
```
setxkbmap fr -variant lafayette -option 'lvl5:ralt_switch_lock'
```
or this command if you also want to swap Esc and Caps Lock:
```
setxkbmap fr -variant lafayette -option 'lvl5:ralt_switch_lock' -option 'caps:swapescape'
```

I wouldn't work on Firefox under Wayland until I started using either of the following env variabales (I changed both for different reasons and can't tell which one did the trick for Lafayette): `MOZ_DBUS_REMOTE=1` and `MOZ_ENABLE_WAYLAND=1`.

It's still not perfect though as I have to start a terminal to run the `setxkbmap` command before I can use the layout and I still don't have access to accented capital letters.

**Comment by taratatach** [comment]
It's not working in firefox anymore today... I don't know how I got it to work yesterday then.

**Comment by maggick** [comment]
Have you tried something like `'xkbcomp -w9 ~/dotfiles/lafayette.xkb $DISPLAY`?

**Comment by taratatach** [comment]
I hadn't tried but just did and the problem remains. From the command name I believe this relates to X11?
I'm running Wayland now and the problem is present only in native Wayland applications so this seems logical.

**Comment by maggick** [comment]
Sorry I miss that part about `Wayland`. Never tried it sorry.

**Comment by stpael** [comment]
I think I got it working on Arch Gnome 40 with Wayland and that it might get it working for other users with Wayland/Gnome in other distribution too.

#### Solving keyboard layout not displayed after install 

First I believe there is a bug with displaying other keyboard layout. When I followed the install described on the [website](https://qwerty-lafayette.org/) it completely screwed up the list of available keyboards. I only saw English US (main keyboard at the time) and French was just empty. 

I add to remove the `type="lafayette"`  tag in both evdev.xml and base.xml (located in /usr/share/X11/xkb/rules/). In other words `<variant type="lafayette">` became `<variant>` in both these files.

I have no idea why this completly breaks the layout lists in the Gnome setting center. I am not sure if this affects other distribution/desktop environment but maybe that is something that can be changed in the install script?

#### Making the Lafayette keyboard layout available in Gnome

After the previous change, keyboard layouts were available again. However, selecting Lafayette layout did not do anything. The solution as suggested [here](https://github.com/fabi1cazenave/qwerty-lafayette/issues/47#issuecomment-379297241) is to first put Lafayette layout as the only layout. Log out and Log In of the Gnome session. You can now add additional layouts and it works like a charm.  THANK YOU <≤èéêûîôâœæà>≥

**Comment by taratatach** [comment]
Thank you so much!! You made my day and the days coming. I had to make a slight adjustment to your solution as the `type` value was `kalamine` in my case but otherwise everything worked.

**Comment by fabi1cazenave** [comment]
Thanks for the work, it should be fixed now with #100.
